### PR TITLE
[ADDED] ObjectStore API in jetstream package

### PR DESCRIPTION
**Pull request by piotrpio**
This PR adds ObjectStore API to `jetstream` package.

It is for the most part a port of `object.go` and `object_test.go` to the new package, without many changes. All object store tests moved from `nats` package pass in the new package (after adding context and changing structs packages)

Notable changes:
- All interfaces and structs ported to `jetstream` package
- `Put()` and `Watch()` methods use the legacy push consumers to prevent permission issues when moving to new API
- API now utilises `context.Context` extensively in line with the rest of the package
- Usage of typed errors

Signed-off-by: Piotr Piotrowski <piotr@synadia.com>

### CODE REVIEW

File: jetstream/object.go [hunk 1]
```diff
@@ -0,0 +1,1297 @@
...
+// ObjectStores is used to retrieve a list of bucket statuses
+func (js *jetStream) ObjectStores(ctx context.Context) ObjectStoresLister {
+	res := &obsLister{
+		obs: make(chan ObjectStoreStatus),
+	}
+	l := &streamLister{js: js}
+	streamsReq := streamsRequest{
+		Subject: fmt.Sprintf(objAllChunksPreTmpl, "*"),
+	}
+	go func() {
```
Comment (Jarema) [inline jetstream/object.go:1241]: We probably should cancel the goroutine in case of `ctx.Done`.

**Review by Jarema — CHANGES_REQUESTED** [review]
Generally looks good and playes nicely with the new `jetstream` package.
There are few things to adresss.

File: jetstream/object.go [hunk 2]
```diff
@@ -0,0 +1,1297 @@
...
+	}
+)
+
+const (
+	objNameTmpl         = "OBJ_%s"     // OBJ_<bucket> // stream name
+	objAllChunksPreTmpl = "$O.%s.C.>"  // $O.<bucket>.C.> // chunk stream subject
+	objAllMetaPreTmpl   = "$O.%s.M.>"  // $O.<bucket>.M.> // meta stream subject
+	objChunksPreTmpl    = "$O.%s.C.%s" // $O.<bucket>.C.<object-nuid> // chunk message subject
+	objMetaPreTmpl      = "$O.%s.M.%s" // $O.<bucket>.M.<name-encoded> // meta message subject
+	objNoPending        = "0"
```
[the remark this answers is not in the document]
Comment (piotrpio) [reply jetstream/object.go:247]: It's a "0", so this means "no pending"

**Review by Jarema — APPROVED** [review]
LGTM!

**Review by Jarema — APPROVED** [review]
LGTM!

**Pull request merged**